dlc/src/lib.rs

```diff
@@ -400,20 +449,65 @@ pub(crate) fn create_fund_transaction_with_fees(
     fund_lock_time: u32,
     fund_output_serial_id: u64,
     extra_fee: u64,
+    fee_percentage_denominator: u64,
+    fee_address: String,
 ) -> Result<(Transaction, Script), Error> {
     let total_collateral = checked_add!(offer_params.collateral, accept_params.collateral)?;
 
-    let (offer_change_output, offer_fund_fee, offer_cet_fee) =
-        offer_params.get_change_output_and_fees(fee_rate_per_vb, extra_fee)?;
-    let (accept_change_output, accept_fund_fee, accept_cet_fee) =
-        accept_params.get_change_output_and_fees(fee_rate_per_vb, extra_fee)?;
+    let offer_change_output = TxOut {
+        value: 0,
+        script_pubkey: offer_params.change_script_pubkey.clone(),
+    };
+
+    let (protocol_fee_amount, protocol_fee_address) =
+        match (fee_percentage_denominator, fee_address) {
+            (fee_denom, fee_address) if fee_denom > 0 && !fee_address.is_empty() => (
+                accept_params.collateral / fee_denom,
+                match Address::from_str(&fee_address) {
+                    Ok(address) => address,
+                    Err(_) => {
+                        return Err(Error::InvalidArgument(
+                            "[create_fund_transaction_with_fees] error: invalid fee address"
+                                .to_string(),
+                        ))
+                    }
+                },
+            ),
+            _ => (
+                0,
+                Address::from_str("1111111111111111111114oLvT2").expect("valid address"),
+            ),
+        };
+
+    println!("desired value for fee: {:?}", (protocol_fee_amount));
+    let acceptor_protocol_fee_output = TxOut {
+        value: protocol_fee_amount,
```

Comment:
I think you need to check if the fee is above dust.

Reply:
I think the outputs under the dust limit get filtered later in the function, under the create_funding_transaction function.